source/Halibut/Transport/Protocol/StreamTimeoutExtensionMethods.cs

```diff
@@ -26,8 +27,20 @@ public static async Task WithTimeout(this Stream stream, SendReceiveTimeout? tim
             }
             finally
             {
-                stream.ReadTimeout = currentReadTimeout;
-                stream.WriteTimeout = currentWriteTimeout;
+                try
+                {
+                    stream.ReadTimeout = currentReadTimeout;
+                    stream.WriteTimeout = currentWriteTimeout;
+                    timeoutsReverted = true;
+                }
+                catch
+                {
+                }
+            }
+
+            if (!timeoutsReverted)
```

Comment:
Is `timeoutsReverted` what we want? 

If `await func()` succeeds, but Read/Write timeout throw an exception, then `timeoutsReverted` will be false, and we will throw an `InvalidOperationException` here. But I thought the point of this PR was to stop throwing an exception in the finally. I.e., aren't we still throwing an exception, just a different one?

I'm likely not understanding something...

Reply:
> This should not happen.

This was a fail safe to make sure it is working as I expected e.g. we throw so it doesn't matter that the timeouts were not reverted. 

I didn't want to swallow errors setting the timeouts when it was a real issue, e.g. the actual call has not thrown, but for some reason the set timeouts throws. That should never happen.

Reply:
> If await func() succeeds, but Read/Write timeout throw an exception
Should this happen? Is it valid? Will it leave us in a weird state if it did happen

Reply:
Ah, I see. The subtilty is that if we set the read/write timeout, and it throws a timeout exception, then this only really happens because it would have already thrown an exception with `await func()`.

Yep, that makes sense 🙂